internal/repo/datasource/redis_test.go

```diff
@@ -0,0 +1,222 @@
+package datasource
+
+import (
+	"errors"
+	"fmt"
+	"github.com/golang/mock/gomock"
+	"github.com/vatsal278/go-redis-cache/mocks"
+	"github.com/vatsal278/html-pdf-service/internal/config"
+	"testing"
+	"time"
+)
+
+func Test_Health(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		setupFunc    func() *mocks.MockCacher
+		validateFunc func(bool)
+	}{
+		{
+			name: "Success::Health Check",
+			setupFunc: func() *mocks.MockCacher {
+				mockcacher := mocks.NewMockCacher(mockCtrl)
+				mockcacher.EXPECT().Health().Times(1).Return("PONG", nil)
+				return mockcacher
+			},
+			validateFunc: func(s bool) {
+				if s != true {
+					t.Errorf("want %v got %v", "not nil", nil)
+				}
+			},
+		},
+		{
+			name: "Failure:: Health Check",
+			setupFunc: func() *mocks.MockCacher {
+				mockcacher := mocks.NewMockCacher(mockCtrl)
+				mockcacher.EXPECT().Health().Times(1).Return("", errors.New(""))
+				return mockcacher
+			},
+			validateFunc: func(s bool) {
+				if s != false {
+					t.Errorf("want %v got %v", "not nil", nil)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockcacher := tt.setupFunc()
+			cacherSvc := config.CacherSvc{Cacher: mockcacher}
+			ds := NewRedisDs(&cacherSvc)
+			x := ds.HealthCheck()
+			tt.validateFunc(x)
+		})
+	}
+
+}
+
+func Test_SaveFile(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		expiry       time.Duration
+		setupFunc    func() *mocks.MockCacher
+		validateFunc func(DataSource, string, error)
```

Comment:
Looks like only the error needs to be validated. The function signature can be updated to only accept the required argument(s).

Reply:
done